test.c

```diff
@@ -0,0 +1,99 @@
+#include "wslib.h"
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <time.h>
+
+void main(int argc, char *argv[]) {
+
+  /*
+  char tmp_name[200];
+  strcpy(tmp_name,"styl.css");
+
+
+  struct safeObject myObject;
+
+  createSafeObjectFromFile(&myObject,tmp_name);
+
+  char *ptr=myObject.sObjectStart;
+  for(int i=0;i<myObject.sSize;i++){
+  printf("%c",*ptr);ptr++;
+  }
+
+  ptr=myObject.sObjectStart;
+  printf("\n----------------\n");
+  for(int i=0;i<myObject.sSize/2;i++){
+  printf("%c",getSafeCharCode(ptr));
+  ptr+=2;}
+
+
+  free(ptr);
+  */
+
+  if (argc == 2) {
+    if (strcmp(argv[1], "-h") == 0) {
+      printf("-h this help\nto output-file source-file\nfrom source-file  "
+             "output-file\nexample:\nWSsafeChars.exe to output.txt myzip.zip  "
+             "--- copies  ");
+      printf("from myzip.zip to  output.txt in safeChars code\nWSsafeChars.exe "
+             "from output.txt mp.zip  --- copies from output.txt to mp.zip");
+    }
+  }
+  if (argc == 4) {
+
+    if (strcmp(argv[1], "to") == 0) {

```

Comment:
to CLI API z `to` i `from` też jest bardziej skomplikowane niż musi być - mogłbyś po prostu robić jak każde inne narzędzie shella, `cp`, `mv`, `ln`, etc. po prostu `program source destination` bez śmiesznych słów kluczowych - bo potem wchodzisz w dziwne edge-casy, że user przekaże tylko `to` a nie `from` - albo na odwrót.

Zgodnie z ideą https://geeklaunch.io/blog/make-invalid-states-unrepresentable/